source/namedCurry.js

```diff
@@ -0,0 +1,53 @@
+import hasPath from './hasPath.js';
+import _filter from './internal/_filter';
+import _curry2 from './internal/_curry2';
+
+/**
+ * Returns a curried equivalent of the provided function with named parameters.
+ * Has all the capabilities of curry, but with named parameters. Supports nested paths as well.
+ * All paths entered as the first argument must be fulfilled before the function is executed.
+ * The following are equivalent, though returned order is not guaranteed:
```

Comment:
I would take out the part here that says "though returned order is not guaranteed". That is really dependent on the behavior of `g` rather than the behavior of this function.